lib/ansible/playbook/conditional.py

```diff
@@ -120,6 +126,12 @@ def _check_conditional(self, conditional, templar, all_vars):
         if conditional in all_vars and re.match("^[_A-Za-z][_a-zA-Z0-9]*$", conditional):
             conditional = all_vars[conditional]
 
+        delimiters = re.search('({{.*}}|{%.*%})', conditional)
```

Comment:
Hrm, I had thought of that, but I cannot find a way to actually use `variable_start_string` inside of a when statement.  Doing so like the following:

```
      when: |
        #jinja2:variable_start_string: "[%" , variable_end_string: "%]"
        [% foo  is defined %]
```

Actually causes us to create a template that looks like:

```
{% if #jinja2:variable_start_string: "[%" , variable_end_string: "%]"
[% foo  is defined %]
 %} True {% else %} False {% endif %}
```

This fails.  Based on this, we don't support changing the delimiters for a when statement.  By the time where we get to where a person could override it in the `Environment` it's already too late.

In any case, I switched to use `templar._clean_data` to check for templating, which removes my regex.